src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -209,18 +226,27 @@ public BoundTransportAddress boundAddress() {
     }
 
     public boolean nodeConnected(DiscoveryNode node) {
-        return transport.nodeConnected(node);
+        return node.id().equals(localNodeId) || transport.nodeConnected(node);
     }
 
     public void connectToNode(DiscoveryNode node) throws ConnectTransportException {
+        if (node.id().equals(localNodeId)) {
+            return;
+        }
         transport.connectToNode(node);
     }
 
     public void connectToNodeLight(DiscoveryNode node) throws ConnectTransportException {
+        if (node.id().equals(localNodeId)) {
+            return;
+        }
         transport.connectToNodeLight(node);
     }
 
     public void disconnectFromNode(DiscoveryNode node) {
+        if (node.id().equals(localNodeId) && localNodeId != null) {
```

Comment:
second null can be removed, yeah.

re this being a noop - yeah, that's how I originally implemented it as well. Then I thought that strictly speaking we don't honor the disconnect (you can still ask connectedNode(localNode) and get true), so we should throw an exception. I think your point valid regarding connetToNodeLight , but in that case, at least currently, we never use a known node id.  Doubting which one is the lesser evil.